engine/src/main/java/io/zeebe/engine/state/deployment/WorkflowPersistenceCache.java

```diff
@@ -50,14 +49,15 @@
   private final PersistedWorkflow persistedWorkflow;
 
   private final ColumnFamily<DbCompositeKey, PersistedWorkflow> workflowByIdAndVersionColumnFamily;
+  private final DbLong workflowVersion;
   private final DbCompositeKey<DbString, DbLong> idAndVersionKey;
 
-  private final ColumnFamily<DbString, DbLong> latestWorkflowColumnFamily;
+  private final ColumnFamily<DbString, LatestWorkflowVersion> latestWorkflowColumnFamily;
   private final DbString workflowId;
-  private final DbLong workflowVersion;
+  private final LatestWorkflowVersion latestVersion = new LatestWorkflowVersion();
 
-  private final ColumnFamily<DbString, DbBuffer> digestByIdColumnFamily;
-  private final DbBuffer digest;
+  private final ColumnFamily<DbString, Digest> digestByIdColumnFamily;
```

Comment:
I tried to do this but it's not very nice since they're used in different contexts it's not very easy to join them. The digest is only read and written during the workflow transformation but the version is often used in the engine so by uniting them we'd be constantly reading the digest unnecessarily. They're also written in different places, the digest is written during the transformation and the version is put in with the deployment. I'm not sure if that needs to stay like that or maybe it can be refactored

Reply:
I don't see a problem that the properties are written and read in different places. I agree that reading the digest if it is not needed, feels not right. On the other side, having another column family may cost a lot of memory. But I'm not sure what is better here. Let's wait on the RocksDB investigation results of Chris.